de.dlr.sc.virsat.graphiti.test/src/de/dlr/sc/virsat/graphiti/diagram/BeanIndependenceSolverTest.java

```diff
@@ -76,91 +88,112 @@ protected void doExecute() {
+		ed1 = new ElementDefinition(conceptEgscc);
+		ed2 = new ElementDefinition(conceptEgscc);
+		sei1 = ed1.getStructuralElementInstance();
+		sei2 = ed2.getStructuralElementInstance();
+		sei1.setUuid(new VirSatUuid(UUID));
+		ed1.add(testDocument2);
 
-		caKey = beanIndependenceSolver.getKeyForBusinessObject(testDocument);
-		assertEquals(DOCUMENTUUID, caKey);
+		command = CreateAddStructuralElementInstanceCommand.create(editingDomain, repository, ed1.getStructuralElementInstance());
+	}
 
-		Command command = CreateAddStructuralElementInstanceCommand.create(editingDomain, repository, ed.getStructuralElementInstance());
-		editingDomain.getCommandStack().execute(command);
+	@Test
+	public void testSolveIndependenceForSei() throws CoreException, IOException {
+		String seiKey = beanIndependenceSolver.getKeyForBusinessObject(ed1);
+		assertEquals(UUID, seiKey);
 
-		editingDomain.getVirSatCommandStack().execute(new RecordingCommand(editingDomain) {
-			@Override
-			protected void doExecute() {
-				rs.getAndAddStructuralElementInstanceResource(sei2);
-			}
-		});
+		editingDomain.getCommandStack().execute(command);
+		getAndAddStructuralElementInstance(sei2);
 
-		Resource resEd2 = rs.getStructuralElementInstanceResource(ed2.getStructuralElementInstance());
-		resEd2.getContents().add(ed2.getStructuralElementInstance());
+		addStructuralElementToResource(ed2);
 		Object sei2Object = beanIndependenceSolver.getBusinessObjectForKey(ed2.getUuid());
 		ABeanStructuralElementInstance aBean = (ABeanStructuralElementInstance) sei2Object;
 		assertEquals(ed2, aBean);
-		
-		editingDomain.getVirSatCommandStack().execute(new RecordingCommand(editingDomain) {
-			@Override
-			protected void doExecute() {
-				rs.getAndAddStructuralElementInstanceResource(sei);
-			}
-		});
 
-		Boolean permission = DiagramHelper.hasDiagramWritePermission(sei);
-		assertEquals(true, permission);
-		
-		Resource resEd = rs.getStructuralElementInstanceResource(ed.getStructuralElementInstance());
-		resEd.getContents().add(ed.getStructuralElementInstance());
+		getAndAddStructuralElementInstance(sei1);
+		checkDiagramWritePermissionForStructuralElement(sei1);
+
+		addStructuralElementToResource(ed1);
 		Object obj = beanIndependenceSolver.getBusinessObjectForKey(seiKey);
 		aBean = (ABeanStructuralElementInstance) obj;
-		assertEquals(ed, aBean);
+		assertEquals(ed1, aBean);
+	}
+
+	@Test
+	public void testSolveIndependenceForBeanProperty() throws CoreException, IOException {
+		getAndAddStructuralElementInstance(sei1);
 
+		//test property
 		Object documentObj = beanIndependenceSolver.getBusinessObjectForKey(testDocument2.getUuid());
 		ABeanCategoryAssignment aCategory = (ABeanCategoryAssignment) documentObj;
 		assertEquals(testDocument2, aCategory);
-		
+
+		Object beanNoteObj = beanIndependenceSolver.getBusinessObjectForKey(testDocument2.getNoteBean().getUuid());
+		BeanPropertyString beanPropertyString = (BeanPropertyString) beanNoteObj;

```

Comment:
Why is this cast here necessary? Cant we just do the the assert on the returned beanNoteObj?

Reply:
We would get a type error if we dont use an explicit cast.

Reply:
But why do we need to cast the beanNoteObj to a BeanPropertyString in the first place?
There shouldnt be any issue doing 

`assertEquals(testDocument2.getNoteBean(), beanNoteObj);`